### PR TITLE
display equivalent odo init command for odo init interactive

### LINKED ISSUE #5465: When running interactive odo init command, get the equivalent command with flags
/kind feature

## Which functionality do you think we should add?

When running `odo init` command in interactive mode, an equivalent command exists with flags. It would be interesting for the user to see the equivalent non interactive command to run:

```
$ odo init
? Select language: go
? Select project type: Go Runtime
 ✓  Downloading devfile "go" from registry "DefaultDevfileRegistry" [927ms]
? Which starter project do you want to use? go-starter
 ✓  Downloading starter project "go-starter" [394ms]
? Enter component name: my-go-app

Your new component "my-go-app" is ready in the current directory.
To start editing your component, use "odo dev" and open this folder in your favorite IDE.
Changes will be directly reflected on the cluster.
To deploy your component to a cluster use "odo deploy".
Next time, you can also run the command:
    odo init --name my-go-app --devfile go --devfile-registry DefaultDevfileRegistry --starter go-starter
```

### PR DESCRIPTION
**Pull request by anandrkskd**
Signed-off-by: anandrkskd <anandrkskd@gmail.com>

/kind feature

**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**

Fixes #5465 

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**
```
run `odo init` in interactive mode
odo init
```

### CODE REVIEW

File: pkg/odo/cli/init/init.go [hunk 1]
```diff
@@ -123,6 +124,13 @@ Your new component '%s' is ready in the current directory.
 To start editing your component, use 'odo dev' and open this folder in your favorite IDE.
 Changes will be directly reflected on the cluster.`, devfileObj.Data.GetMetadata().Name)
 
+	automateCommad := fmt.Sprintf("\nPort configuration using flag is currently not supported  \n\nYou can automate this command by executing:\n   odo init --name %s --devfile %s --devfile-registry %s", name, devfileLocation.Devfile, devfileLocation.DevfileRegistry)
+	if len(o.flags) == 0 && starterInfo != nil {
+		log.Infof("%s --starter %s", automateCommad, starterInfo.Name)
+	} else if len(o.flags) == 0 {
+		log.Infof("%s", automateCommad)
+	}
```
Comment (feloy) [inline pkg/odo/cli/init/init.go:132]: I would separate the build of the comamnd string from the rest of the message.

Also, the condition of len(o.flags) could be placed n a better place.

I think you could instead write something like:

```
if len(o.flags) == 0 {
  automateCommad := fmt.Sprintf(odo init --name %s --devfile %s --devfile-registry %s", ...)
  if starterInfo != nil {
    automateCommand = fmt.Sprintf("%s --starter %s", automateCommad, starterInfo.Name
  }
  log.Infof("\nPort configuration using flag is currently not supported  \n\nYou can automate this command by executing:\n   %s\n", automateCommad)
}

File: pkg/odo/cli/init/init.go [hunk 2]
```diff
@@ -147,7 +155,7 @@ func (o *InitOptions) RunForJsonOutput(ctx context.Context) (out interface{}, er
 }
 
 // run downloads the devfile and starter project and returns the content and the path of the devfile
```
Comment (feloy) [inline pkg/odo/cli/init/init.go:157]: Please adapt the documentation to add the new returned values

**Review by feloy — CHANGES_REQUESTED** [review]

**Comment by feloy** [comment]
@anandrkskd could you please rebase this PR?
e2e tests fail on some issue fixed by a PR merged recently

**Comment by anandrkskd** [comment]
Thanks Philippe, rebased the PR.

**Pull request merged**